### PR TITLE
[HttpFoundation] Add tests for uncovered sections

### PR DESCRIPTION
**Pull request by alexandre-daubois**
| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Issues        | -
| License       | MIT

### CODE REVIEW

**Comment by OskarStark** [comment]
Great, I propose to merge such PRs in lower branches. 😎

**Comment by alexandre-daubois** [comment]
That's what I went for at first but I remembered that this topic was raised in my other PRs about test coverage, where it's merged in the highest branch possible 😄 For example: https://github.com/symfony/symfony/pull/52771#issuecomment-1905597295, https://github.com/symfony/symfony/pull/52746#pullrequestreview-1754583939

**Comment by OskarStark** [comment]
Yes, same for me 😄

File: src/Symfony/Component/HttpFoundation/Tests/RequestMatcher/QueryParameterRequestMatcherTest.php [hunk 1]
```diff
@@ -13,6 +13,7 @@
 
 use PHPUnit\Framework\TestCase;
 use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\RequestMatcher\HeaderRequestMatcher;
```
Comment (OskarStark) [inline src/Symfony/Component/HttpFoundation/Tests/RequestMatcher/QueryParameterRequestMatcherTest.php]: Can be removed?

Comment (alexandre-daubois) [reply src/Symfony/Component/HttpFoundation/Tests/RequestMatcher/QueryParameterRequestMatcherTest.php]: Indeed thanks! I thought fabbot would have warn me about it :thinking:

**Comment by derrabus** [comment]
There's nothing wrong with increasing test coverage on LTS branches. I'd say, go for it.

**Comment by alexandre-daubois** [comment]
Suits me fine. I opened a follow-up for 6.4 and will do another for 7.1 for the remaining ones if this gets actually merged in 5.4

**Review by derrabus — APPROVED** [review]

File: src/Symfony/Component/HttpFoundation/Tests/RateLimiter/AbstractRequestRateLimiterTest.php [hunk 2]
```diff
@@ -33,6 +33,26 @@ public function testConsume(array $rateLimits, ?RateLimit $expected)
         $this->assertSame($expected, $rateLimiter->consume(new Request()));
     }
 
+    public function testConsumeWithoutLimiterAddSpecialNoLimiter()
```
Comment (derrabus) [inline src/Symfony/Component/HttpFoundation/Tests/RateLimiter/AbstractRequestRateLimiterTest.php]: This test fails on the low-deps run: https://github.com/symfony/symfony/actions/runs/9989450887/job/27608100847#step:7:1761

Comment (alexandre-daubois) [reply src/Symfony/Component/HttpFoundation/Tests/RateLimiter/AbstractRequestRateLimiterTest.php]: This is an underlying bug in 5.2.0, where `\INF` was used before being changed to `\PHP_INT_MAX`: https://github.com/symfony/symfony/blob/v5.2.0/src/Symfony/Component/RateLimiter/Policy/NoLimiter.php#L37. I skip the test when NoLimiter is buggy. What do you think?

File: src/Symfony/Component/HttpFoundation/Tests/RateLimiter/AbstractRequestRateLimiterTest.php [hunk 3]
```diff
@@ -33,6 +34,30 @@ public function testConsume(array $rateLimits, ?RateLimit $expected)
...
     }
 
+    public function testConsumeWithoutLimiterAddsSpecialNoLimiter()
+    {
+        $rateLimiter = new MockAbstractRequestRateLimiter([]);
+
+        try {
+            $this->assertSame(\PHP_INT_MAX, $rateLimiter->consume(new Request())->getLimit());
+        } catch (\TypeError $error) {
+            $this->markTestSkipped('This test cannot be run on a version of the RateLimiter component that uses \INF instead of \PHP_INT_MAX in NoLimiter.');
```
Comment (nicolas-grekas) [inline src/Symfony/Component/HttpFoundation/Tests/RateLimiter/AbstractRequestRateLimiterTest.php]: This would swallow any TypeError. Not a good thing.

Comment (alexandre-daubois) [reply src/Symfony/Component/HttpFoundation/Tests/RateLimiter/AbstractRequestRateLimiterTest.php]: Right, I added a check on the error message. If I remember correctly, something like this has already been done elsewhere in the code.

**Comment by nicolas-grekas** [comment]
Thank you @alexandre-daubois.

**Pull request merged**